rs/index/src/posting_list/file.rs

```diff
@@ -0,0 +1,673 @@
+                    buffer.clear();
+                }
+            }
+        }
+        // Write any remaining data in the buffer after processing all posting lists
+        if !buffer.is_empty() {
+            self.write_to_current_mmap(&buffer)?;
+            buffer.clear();
+        }
+        self.resident = false;
+        self.resident_posting_lists.clear();
+        Ok(())
+    }
+
+    // The caller is responsible for setting self.current_offset to the right value
+    fn append_posting_list_to_disk(&mut self, posting_list: &[usize]) -> Result<()> {
+        if self.resident {
+            return Err(anyhow!("Posting lists should already be flushed to disk"));
+        }
+        let usize_in_bytes = std::mem::size_of::<usize>();
+        // First write the metadata
+        if self.current_offset == self.backing_file_size {
+            self.new_backing_file()?;
+        }
+        let mut buffer: Vec<u8> = vec![];
+        buffer.extend_from_slice(posting_list.len().to_le_bytes().as_ref());
+        buffer.extend_from_slice(self.offset_to_current_posting_list.to_le_bytes().as_ref());
+        self.write_to_current_mmap(&buffer)?;
+        buffer.clear();
+
+        // Now write the posting list
+        self.current_offset = self.offset_to_current_posting_list;
+        self.offset_to_current_posting_list += posting_list.len() * usize_in_bytes;
+        for idx in posting_list.iter() {
+            if self.current_offset == self.backing_file_size {
+                self.new_backing_file()?;
+            }
+            buffer.extend_from_slice(idx.to_le_bytes().as_ref());
+            // Write as much as possible to the current backing file
+            let available_space = self.backing_file_size - self.current_offset;
+            if buffer.len() == available_space {
+                self.write_to_current_mmap(&buffer)?;
+                buffer.clear();
+            }
+        }
+        // Write any remaining data in the buffer after processing all posting lists
+        if !buffer.is_empty() {
+            self.write_to_current_mmap(&buffer)?;
+            buffer.clear();
+        }
+        self.current_num_of_posting_list += 1;
+        Ok(())
+    }
+
+    fn offset_to_file_num_and_file_offset(&self, offset: usize) -> Result<FileAccessId> {
+        let file_num = offset / self.backing_file_size;
+        if file_num >= self.mmaps.len() {
+            return Err(anyhow!("File number out of bound"));
+        }
+
+        let file_offset = offset % self.backing_file_size;
+        if file_offset >= self.mmaps[file_num].len() {
+            return Err(anyhow!("File offset out of bound"));
+        }
+
+        Ok(FileAccessId {
+            file_num,
+            file_offset,
+        })
+    }
+}
+
+impl PostingListStorage for FileBackedAppendablePostingListStorage {
+    fn get(&self, id: u32) -> Result<Vec<usize>> {
+        let i = id as usize;
+        let usize_in_bytes = std::mem::size_of::<usize>();
```

Comment:
this is actually dangerous - `usize` is not fixed size in all architecture. If you plan to use it for offset, make sure you change it to `u64`

Reply:
https://doc.rust-lang.org/reference/type-layout.html#primitive-data-layout